magento/pwa-studio
# getNavigationMenu Query Efficiency

## Description

In our `getNavigationMenu` query, [there is a comment](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/queries/getNavigationMenu.graphql#L11-L12) that explains an inefficiency in the query:

>      # Temporarily include products field in query to get around GraphQL bug
>      # affecting Magento Cloud Pro instances. Remove once magento\graphql-ce\#374 is fixed.

**tl;dr:**

There is a new `products` field but it just issues a `products` query and it doesn't have any way to limit the size of the response (the default is 20 items).

We can improve the efficiency of this default behavior if we keep our `products` query with the `pageSize` set to `1`. 

**More Details:**

I tracked down https://github.com/magento/graphql-ce/issues/374 and found it was closed and a fix supposedly went in with https://github.com/magento/graphql-ce/issues/567.

There is now a `products` field in `category` that gives us access to the products of a category without having to issue a `products` query as mentioned in the comment.

The new query would be:

```graphql
query navigationMenu($id: Int!) {
  category(id: $id) {
    id
    name
    children {
      children_count
      id
      include_in_menu
      name
      position
      productImagePreview: products {
        items {
          small_image{
            url
          }
        }
      }
      url_path
    }
  }
}
```

But unfortunately there is no way to limit the size of the `products.items` array that is returned. So if we update our query, we will get the `small_image.url` of the first 20 items (the default limit) in the category. 

All we ever use is [the first one](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/CategoryList/useCategoryTile.js#L19), so we'd actually be drastically overfetching data if we made this change.

Since it appears the `CategoryInterface` includes a `products` sub-query anyway, we're also not extremely deviating from the functionality already in place:

```graphql
interface CategoryInterface {
  # other fields
  products(
    pageSize: Int = 20
    currentPage: Int = 1
    sort: ProductSortInput
  ): CategoryProducts
}
```

So long story short: There is a new `products` field but it just issues a `products` query and it doesn't have any way to limit the size of the response (the default is 20 items).

We can improve the efficiency of this default behavior if we keep our `products` query with the `pageSize` set to `1`. 

## Related Issue





Closes #1677 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps



Check my reasoning on why we can safely remove this comment and no further changes are needed.
There are no steps to verify anything, as the resolution of this ticket is simply removing a `TODO` comment from the codebase.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
